plugin/trino-snowflake/src/test/java/io/trino/plugin/snowflake/TestSnowflakeConnectorTest.java

```diff
@@ -0,0 +1,383 @@
+    {
+        // Override and skip it because snowflake not support this feature
+        // Unsupported delete
+        assertThatThrownBy(super::testDeleteWithVarcharEqualityPredicate);
+    }
+
+    @Test
+    @Override
+    public void testDeleteWithVarcharGreaterAndLowerPredicate()
+    {
+        // Override and skip it because snowflake not support this feature
+        // Unsupported delete
+        assertThatThrownBy(super::testDeleteWithVarcharGreaterAndLowerPredicate);
+    }
+
+    @Test
+    @Override
+    public void testDeleteWithVarcharInequalityPredicate()
+    {
+        // Override and skip it because snowflake not support this feature
+        // Unsupported delete
+        assertThatThrownBy(super::testDeleteWithVarcharInequalityPredicate);
+    }
+
+    @Test
+    @Override
+    public void testInsertInPresenceOfNotSupportedColumn()
+    {
+        // Override and skip it because snowflake not support this feature
+        // Invalid number precision: 50. Must be between 0 and 38.
+        assertThatThrownBy(super::testInsertInPresenceOfNotSupportedColumn);
+    }
+
+    @Test
+    @Override
+    public void testAggregationPushdown()
+    {
+        // Override and skip it because snowflake not support this feature
+        assertThatThrownBy(super::testAggregationPushdown).isInstanceOf(AssertionError.class);
+    }
+
+    @Test
+    @Override
+    public void testDistinctAggregationPushdown()
+    {
+        // Override and skip it because snowflake not support this feature
+        assertThatThrownBy(super::testDistinctAggregationPushdown).isInstanceOf(AssertionError.class);
+    }
+
+    @Test
+    @Override
+    public void testNumericAggregationPushdown()
+    {
+        // Override and skip it because snowflake not support this feature
+        assertThatThrownBy(super::testNumericAggregationPushdown).isInstanceOf(AssertionError.class);
+    }
+
+    @Test
+    @Override
+    public void testLimitPushdown()
+    {
+        // Override and skip it because snowflake not support this feature
+        assertThatThrownBy(super::testLimitPushdown).isInstanceOf(AssertionError.class);
+    }
+
+    @Test
+    @Override
+    public void testInsertIntoNotNullColumn()
+    {
+        // Override and skip it because snowflake not support this feature
+        assertThatThrownBy(super::testInsertIntoNotNullColumn);
+    }
+
+    @Test
+    @Override
+    public void testPotentialDuplicateDereferencePushdown()
+    {
+        // Override and skip it because snowflake not support this feature
+        assertThatThrownBy(super::testPotentialDuplicateDereferencePushdown);
+    }
+
+    @Test
+    @Override
+    public void testDeleteWithLike()
+    {
+        // Override and skip it because snowflake not support this feature
+        assertThatThrownBy(super::testDeleteWithLike).isInstanceOf(AssertionError.class);
+    }
+
+    @Test
+    @Override
+    public void testNativeQueryCreateStatement()
```

Comment:
Add a below line to `SnowflakeClientModule` and remove all `testNative*` test methods. I suppose `testNativeQueryCreateStatement` & `testNativeQueryInsertStatementTableExists` still require override. 
```java
        newSetBinder(binder, ConnectorTableFunction.class).addBinding().toProvider(Query.class).in(Scopes.SINGLETON);

```

Reply:
Added